src/cpu.c

```diff
@@ -619,47 +622,140 @@ static void ADD_A_n(uint8_t n) {
 void OPC_ADD_A_A(void) {
     ADD_A_n(CPU_REG_A);
     cpu.cycle_count += 4;
+    ++cpu.PC;
 }
 
 void OPC_ADD_A_B(void) {
     ADD_A_n(CPU_REG_B);
     cpu.cycle_count += 4;
+    ++cpu.PC;
 }
 
 void OPC_ADD_A_C(void) {
     ADD_A_n(CPU_REG_C);
     cpu.cycle_count += 4;
+    ++cpu.PC;
 }
 
 void OPC_ADD_A_D(void) {
     ADD_A_n(CPU_REG_D);
     cpu.cycle_count += 4;
+    ++cpu.PC;
 }
 
 void OPC_ADD_A_E(void) {
     ADD_A_n(CPU_REG_E);
     cpu.cycle_count += 4;
+    ++cpu.PC;
 }
 
 void OPC_ADD_A_H(void) {
     ADD_A_n(CPU_REG_H);
     cpu.cycle_count += 4;
+    ++cpu.PC;
 }
 
 void OPC_ADD_A_L(void) {
     ADD_A_n(CPU_REG_L);
     cpu.cycle_count += 4;
+    ++cpu.PC;
 }
 
 void OPC_ADD_A_HL(void) {
     ADD_A_n(mmu_get_byte(CPU_DREG_HL));
     cpu.cycle_count += 8;
+    ++cpu.PC;
 }
 
 void OPC_ADD_A_d8(void) {
     uint8_t immediate = mmu_get_byte(cpu.PC + 1);
     ADD_A_n(immediate);
     cpu.cycle_count += 8;
+    cpu.PC += 2;
+}
+
+// GCC complains about the rhs of "uint_fast16_t result = A + n + c_flag" being a _signed_ int
```

Comment:
An alternative way to handle this is to explicitly cast the result, maybe an alternative?
On the other hand this is more explicit in stating the underlying problem, so I'm not sure

Reply:
Yes, that should fix it 🤔 
Although I have to agree with the last thing you said. It is more explicit and does not "hide" the bug.

I'm open for both, you tell me what you prefer.